### ISSUE TITLE
VS Code server process not being closed/cleaned in a timely manner

### ISSUE THREAD

**Issue by tanhakabir** [microsoft/vscode-remote-release#6163]
https://github.com/microsoft/vscode-remote-release/issues/5955#issuecomment-1011485902

**Comment by tanhakabir** [comment]
cc @PatMan8

**Comment by xendren** [comment]
Any thoughts on when this might be fixed?

**Comment by tanhakabir** [comment]
Sorry I realized this is a duplicate of this feature request: https://github.com/microsoft/vscode-remote-release/issues/440

Which has more details about why VS Code Server process closes after 8 hours.

There's this suggestion we can look into experimenting with in the next iteration (March) https://github.com/microsoft/vscode-remote-release/issues/6281#issuecomment-1046770198

**Comment by xendren** [comment]
> Sorry I realized this is a duplicate of this feature request: #440
> 
> Which has more details about why VS Code Server process closes after 8 hours.
> 
> There's this suggestion we can look into experimenting with in the next iteration (March) [#6281 (comment)](https://github.com/microsoft/vscode-remote-release/issues/6281#issuecomment-1046770198)

Ok.  My original complaint was that the vs code servers were caching GB's of data per user on our dev server.  That started with an update back in early 2021.  So it seems there is no solution for that complaint.  The assumption is that the servers should at least clean up after themselves eventually with the feature request #440?

**Comment by tanhakabir** [comment]
@xendren they do clean themselves up after 8 hours currently. #440 is asking for that to be configurable to be longer or much shorter. How often do you see that your dev servers are being overloaded with VS Code Server processes?

**Comment by xendren** [comment]
> @xendren they do clean themselves up after 8 hours currently. #440 is asking for that to be configurable to be longer or much shorter. How often do you see that your dev servers are being overloaded with VS Code Server processes?

There are issues that we have encountered.  
1. The server extensions seem to cache large amounts of data in the user's home directory as of an update early 2021.  These caches are multiple GBs per user, so we have space issues that come up frequently.
2. There was a bug where server processes were not being cleaned up after shutdown.

We reported #1 in a previous issue.  During discussions on that one, we discovered issue #2.  I believe issue #2 has been fixed.  Issue #1 is still a problem for us since we have dev servers with limited disk space.  At some point over the past year, I saw a recommendation to add a setting that would limit the size of the cache the extension servers could use, but that did not move any further than a suggestion, I believe.  I'm guessing this is just something we have to live with since #440 wouldn't address the cache size problem.
